spec/category_spec.rb

```diff
@@ -828,6 +828,22 @@
         expect(@dv[:yoda]).to eq(666)
       end
 
+      it "assigns a new category at a new index " do
+        @dv.add_category :abc
+        @dv.send(:set, :xtbr, :abc)
+        expect(@dv[:xtbr]).to eq(:abc)
+      end
+
+      it "assigns a new category at a multiple indices " do
+        @dv.add_category :rbn
+          @dv.send(:set, [:padme, :bdn, :cy..:db], :rbn)
+        expect(@dv[:bdn]).to eq(:rbn)
```

Comment:
The iterators aren't supposed to be supported inside an array. What I meant was that they can be used like this `dv.set 1..4, :abc`. This is equivalent to `dv.set [1, 2, 3, 4], :abc`

Reply:
Okay, I have done changes which can handle arrays and iterators at the same time, can we add this functionality to the method in vector.rb as well? Also, you have been calling `set` as a public method, does that change need to be done too?

Reply:
I think method in vector.rb already has this functionality. No there's no need to change the method to public.

Reply:
Please check this [notebook](http://nbviewer.jupyter.org/gist/rohitner/1d1223d29338715d59f0b91532595544). There is also a bug which can be seen in the last query. I hope there aren't any other changes needed before this could be merged.

Reply:
You need not support array and range at the same time. So it's not a bug. The user can either use an array, single element, range or iterator but not a mixture of these at the same time.

Reply:
I have dropped support for a mixture of arguments, please check.